src/main/java/org/marketplace/services/AdvertisementImageManagementService.java

```diff
@@ -1,26 +1,39 @@
 package org.marketplace.services;
 
+import jakarta.persistence.EntityExistsException;
 import jakarta.persistence.EntityNotFoundException;
 import org.marketplace.models.AdvertisementImage;
 import org.marketplace.repositories.AdvertisementImageManagementRepository;
 import org.springframework.stereotype.Service;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.acls.domain.BasePermission;
+import org.springframework.security.acls.model.MutableAcl;
+import org.springframework.security.acls.model.MutableAclService;
+import org.springframework.security.acls.domain.ObjectIdentityImpl;
+import org.springframework.security.acls.domain.PrincipalSid;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.transaction.annotation.Transactional;
 
 import java.util.List;
 
 @Service
 public class AdvertisementImageManagementService {
     private final AdvertisementImageManagementRepository advertisementImageManagementRepository;
 
-    public AdvertisementImageManagementService(AdvertisementImageManagementRepository advertisementImageManagementRepository) {
+    private final MutableAclService aclService;
+
+    public AdvertisementImageManagementService(AdvertisementImageManagementRepository advertisementImageManagementRepository, MutableAclService aclService) {
         this.advertisementImageManagementRepository = advertisementImageManagementRepository;
+        this.aclService = aclService;
     }
 
     public AdvertisementImage addImage(AdvertisementImage advertisementImage) {
         try {
-            getAdvertisementImageById(advertisementImage.getId());
-            throw new EntityNotFoundException(String.format("Advertisement image with id: %d already exists!", advertisementImage.getId()));
+            //getAdvertisementImageById(advertisementImage.getId());
+            //throw new EntityExistsException(String.format("Advertisement image with id: %d already exists!", advertisementImage.getId()));
+            return save(advertisementImage);
```

Comment:
sure, i added this one because it always did throw exception for me and was not sure about intended logic
so we might want to add like 
```if (getImgById!=null){throw new SomeException("some error");}```